docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -2,33 +2,605 @@
+                
+                canister_ids = vec {principal "ltyfs-qiaaa-aaaak-aan3a-cai", principal "ltyfs-qiaaa-aaaak-aan3a-cai"};
+                
+                new_controllers = vec {principal "rymrc-piaaa-aaaao-aaljq-cai", principal "suaf3-hqaaa-aaaaf-bfyob-cai"};
+            };
+        };
+    };
+)' > message.json
+
+quill send message.json
+```
+
+## `TransferSnsTreasuryFunds`
+
+The SNS DAO has control over a treasury from which funds can be sent to other accounts by `TransferSnsTreasuryFunds` proposals.
+To do so, one has to define the following arguments... 
+
+### Relevant type signatures
+
+```candid
+    type TransferSnsTreasuryFunds = record {
+        from_treasury : int32;
+        to_principal : opt principal;
+        to_subaccount : opt Subaccount;
+        memo : opt nat64;
+        amount_e8s : nat64;
+    };
+
+    type Subaccount = record { subaccount : vec nat8 };
+```
+
+### Putting it together
+
+```candid
+    type TransferSnsTreasuryFunds = record {
+        from_treasury : int32;
+        to_principal : opt principal;
+        to_subaccount : opt record { subaccount : vec nat8 };
+        memo : opt nat64;
+        amount_e8s : nat64;
+    };
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "Transfer 41100 ICP to Foo Labs";
+        url = "https://sns-examples.com/proposal/42";
+        summary = "Transfer 411 ICP to Foo Labs";
+        action = opt variant {
+            TransferSnsTreasuryFunds = record {
+
+                from_treasury = 1 : int32;
+                
+                to_principal = opt principal "ozcnp-xcxhg-inakz-sg3bi-nczm3-jhg6y-idt46-cdygl-ebztx-iq4ft-vae";
+                
+                to_subaccount = null;
+                
+                memo = null;
+                
+                amount_e8s = 4_110_000_000_000 : nat64;
+            };
+        };
+    };
+)' > message.json
+
+quill send message.json
+```
+
+See example [proposal of an active SNS](https://dashboard.internetcomputer.org/sns/3e3x2-xyaaa-aaaaq-aaalq-cai/proposal/202).
+
+## `UpgradeSnsControlledCanister` 
+
+The proposal `UpgradeSnsControlledCanister` is to upgrade a dapp canister that is controlled by the SNS DAO to a new wasm. 
+
+### Relevant type signatures
+
+```candid
+    type UpgradeSnsControlledCanister : UpgradeSnsControlledCanister;
+
+    type UpgradeSnsControlledCanister = record {
+        new_canister_wasm : vec nat8;
+        mode : opt int32;
+        canister_id : opt principal;
+        canister_upgrade_arg : opt vec nat8;
+    };
+```
+
+:::warning
+Because this proposal requires passing Wasm which is unwieldly to copy/paste into the commandline as binary. It is recommended that developers use the specially-made [`make-upgrade-canister-proposal`](https://github.com/dfinity/quill/blob/master/docs/cli-reference/sns/quill-sns-make-upgrade-canister-proposal.md) command in `quill sns`.
```

Comment:
```suggestion
Because this proposal requires passing wasm, which is unwieldy to copy/paste into the command line as binary, it is recommended that developers use the specially-made [`make-upgrade-canister-proposal`](https://github.com/dfinity/quill/blob/master/docs/cli-reference/sns/quill-sns-make-upgrade-canister-proposal.md) command in `quill sns`.
```